openassessment/xblock/openassessmentblock.py

```diff
@@ -1324,6 +1341,36 @@ def publish_assessment_event(self, event_name, assessment, **kwargs):
             event_data
         )
 
+    @XBlock.json_handler
+    def reset_submission(self, data, suffix=""):  # pylint: disable=unused-argument
+        """
+        Reset the student's submission.
+
+        Args:
+            data (dict): Unused parameter. Defaults to {}.
+            suffix (str, optional): Unused parameter. Defaults to ''.
+
+        Returns:
+            dict: A dictionary indication the status with keys 'success' (bool) and 'msg' (str)
+        """
+        StudentModule = import_student_module()
+        User = get_user_model()
+
+        block_user = self.runtime.service(self, "user").get_current_user()
+        username = block_user.opt_attrs.get("edx-platform.username")
+
```

Comment:
Since our UI isn't dynamically updated, if the learner keeps their submission page open, it's possible for them to reset their submission even after staff have started grading.

Is it possible to check `allow_resubmission` here before actually doing the reset?

Reply:
With the previous change I was getting an error when resetting the submission. Now it should work with the changes made in this commit: https://github.com/openedx/edx-ora2/commit/4b944bbd4ca3c8a718ef099a621e95e9eac630d2

Reply:
Thanks for that! It's working well with this change.